tests/test_decorators.py

```diff
@@ -518,3 +518,33 @@ def validate_many(self, data, many):
         errors = schema.validate([{'foo': 3, 'bar': 'not an int'}], many=True)
         assert 'bar' in errors[0]
         assert '_schema' not in errors
+
+    def test_allow_reporting_field_errors_in_schema_validator(self):
+
+        class NestedSchema(Schema):
+            baz = fields.Int(required=True)
+
+        class MySchema(Schema):
+            foo = fields.Int(required=True)
+            bar = fields.Nested(NestedSchema, required=True)
+            bam = fields.Int(required=True)
+
+            @validates_schema(skip_on_field_errors=True)
+            def consistency_validation(self, data):
+                errors = {}
+                if data['bar']['baz'] != data['foo']:
+                    errors['bar'] = {'baz': 'Non-matching value'}
+
+                if data['bam'] > data['foo']:
+                    errors['bam'] = 'Value should be less than foo'
+
+                if errors:
+                    raise ValidationError(errors)
+
+        schema = MySchema()
+        errors = schema.validate({'foo': 2, 'bar': {'baz': 5}, 'bam': 6})
```

Comment:
As I mentioned before (in [original issue description](issues/441)), the test case that I modified look like that feature (using non-string structure as error message) was accidental. The test itself aims to verify completely different aspect of the library (passing original data to validator). If there would be a special test that checked that it is possible to use non-string error messages than fine. But there is no such test case.

As of breaking validator into separate validators, this is not acceptable for me:
1. You still can not report errors for nested fields (only fields of top level object)
2. You can not report errors for list elements (since you can not make library re-run validator multiple times, reporting one error at a time).